File: backend/app/schemas/test_case_version.py
"""
测试用例版本相关的 Pydantic 模型
"""
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime


class TestCaseVersionBase(BaseModel):
    """测试用例版本基础模型"""
    test_case_id: int = Field(..., description="测试用例ID")
    version: str = Field(..., min_length=1, max_length=50, description="版本号")
    name: Optional[str] = Field(None, max_length=200, description="版本名称")
    description: Optional[str] = Field(None, description="版本备注")
    content: Optional[Dict[str, Any]] = Field(None, description="版本内容（完整的测试用例数据快照）")


class TestCaseVersionCreate(TestCaseVersionBase):
    """创建测试用例版本模型"""
    pass


class TestCaseVersionUpdate(BaseModel):
    """更新测试用例版本模型"""
    name: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None


class TestCaseVersionResponse(TestCaseVersionBase):
    """测试用例版本响应模型"""
    id: int
    created_by: Optional[int] = None
    created_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True

File: frontend/src/store/services/testCaseVersion.ts
import { api } from './api'

export interface TestCaseVersion {
  id: number
  test_case_id: number
  version: string
  name?: string
  description?: string
  content?: Record<string, any>
  created_by?: number
  created_at?: string
}

export interface TestCaseVersionCreate {
  test_case_id: number
  version: string
  name?: string
  description?: string
  content?: Record<string, any>
}

export interface TestCaseVersionUpdate {
  name?: string
  description?: string
}

export interface TestCaseVersionListParams {
  test_case_id?: number
  skip?: number
  limit?: number
}

export interface VersionCompareResult {
  version_id: number
  version: string
  compare_with: string | number
  differences: Record<string, {
    version: any
    compare: any
  }>
}

export const testCaseVersionService = {
  async getVersions(params?: TestCaseVersionListParams): Promise<TestCaseVersion[]> {
    const response = await api.get<TestCaseVersion[]>('/test-case-versions', { params })
    return response.data
  },

  async getVersion(id: number): Promise<TestCaseVersion> {
    const response = await api.get<TestCaseVersion>(`/test-case-versions/${id}`)
    return response.data
  },

  async createVersion(data: TestCaseVersionCreate): Promise<TestCaseVersion> {
    const response = await api.post<TestCaseVersion>('/test-case-versions', data)
    return response.data
  },

  async updateVersion(id: number, data: TestCaseVersionUpdate): Promise<TestCaseVersion> {
    const response = await api.put<TestCaseVersion>(`/test-case-versions/${id}`, data)
    return response.data
  },

  async deleteVersion(id: number): Promise<void> {
    await api.delete(`/test-case-versions/${id}`)
  },

  async restoreVersion(id: number): Promise<{ message: string; test_case_id: number }> {
    const response = await api.post<{ message: string; test_case_id: number }>(`/test-case-versions/${id}/restore`)
    return response.data
  },

  async compareVersions(version_id: number, compare_with?: number): Promise<VersionCompareResult> {
    const response = await api.get<VersionCompareResult>(`/test-case-versions/${version_id}/compare`, {
      params: compare_with ? { compare_with } : {}
    })
    return response.data
  },
}

